service/core/v2ray/asset/asset.go

```diff
@@ -1,86 +1,122 @@
 package asset
 
 import (
+	"errors"
+	"github.com/adrg/xdg"
 	"github.com/muhammadmuzzammil1998/jsonc"
-	"github.com/v2rayA/v2rayA/common"
 	"github.com/v2rayA/v2rayA/common/files"
 	"github.com/v2rayA/v2rayA/conf"
 	"github.com/v2rayA/v2rayA/core/v2ray/where"
 	"github.com/v2rayA/v2rayA/pkg/util/log"
+	"io/fs"
 	"os"
 	"path"
 	"path/filepath"
+	"runtime"
 	"time"
 )
 
-func GetV2rayLocationAsset() (s string) {
-	var candidates = []string{
-		"/usr/local/share/v2ray",
-		"/usr/share/v2ray",
-		"/opt/share/v2ray",
-		"/usr/local/share/xray",
-		"/usr/share/xray",
-		"/opt/share/xray",
+func GetV2rayLocationAssetOverride() string {
+	if runtime.GOOS != "windows" {
+		return filepath.Join(xdg.RuntimeDir, "v2raya")
+	} else {
+		return GetV2rayConfigDirPath()
```

Comment:
This is the value of --v2ray-confdir. Do you mean `conf.GetEnvironmentConfig().Config` ? The same below.